tests/cli_test.py

```diff
@@ -125,12 +127,267 @@ def cleos_sign_test():
+    cmd = ['./programs/cleos/cleos', '-u','http://127.0.0.1:12345', '--abi-file', system_abi_file_arg, token_abi_file_arg, 'convert', 'pack_transaction', '--pack-action-data', unpacked_trx]
+    outs, errs = processCleosCommand(cmd)
+    assert(expected_output in outs)
+
+    # unpack transaction
+    packed_trx = """{
+        "signatures": [
+            "SIG_K1_K3LfbB7ZV2DNBu67iSn3yUMseTdiwoT49gAcwSZVT1QTvGXVHjkcvKqhentCW4FJngZJ1H9gBRSWgo9UPiWEXWHyKpXNCZ"
+        ],
+        "compression": "none",
+        "packed_context_free_data": "",
+        "packed_trx": "3aacf360ee010b864b7e00000000020000000000ea305500409e9a2264b89a010000000000ea305500000000a8ed3232660000000000ea30550000000000000e3d01000000010002c0ded2bc1f1305fb0faac5e6c03ee3a1924234985427b6167ca569d13df435cf0100000001000000010002c0ded2bc1f1305fb0faac5e6c03ee3a1924234985427b6167ca569d13df435cf0100000000a6823403ea3055000000572d3ccdcd010000000000008c3100000000a8ed3232260000000000008c31000000000000ce39a08601000000000004535953000000000568656c6c6f00"
+    }"""
+    cmd = ['./programs/cleos/cleos', '-u','http://127.0.0.1:12345', '--abi-file', system_abi_file_arg, token_abi_file_arg, 'convert', 'unpack_transaction', '--unpack-action-data', packed_trx]
+    outs, errs = processCleosCommand(cmd)
+    assert(b'"creator": "eosio"' in outs)
+    assert(b'"name": "bob"' in outs)
+
+    assert(b'"from": "aaa"' in outs)
+    assert(b'"to": "bbb"' in outs)
+    assert(b'"quantity": "10.0000 SYS"' in outs)
+    assert(b'"memo": "hello"' in outs)
+
+    # push action token transfer with option `--abi-file`
+    try:
+        data_dir = "./taf_data"
+        cmd = "./programs/nodeos/nodeos -e -p eosio --plugin eosio::producer_plugin --plugin eosio::producer_api_plugin --plugin eosio::chain_api_plugin --plugin eosio::chain_plugin --plugin eosio::http_plugin --access-control-allow-origin=* --http-validate-host=false " + "--data-dir " + data_dir
+        pNodeos = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        time.sleep(10)
+
+        PUBLIC_KEY  = "EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV"
+        PRIVATE_KEY = "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3"
+
+        wallet_name = "taf_wallet"
+        cmd = "./programs/cleos/cleos wallet create --name " + wallet_name + " --to-console"
+        pWallet = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        cmd = "./programs/cleos/cleos wallet import --name " + wallet_name + " --private-key " + PRIVATE_KEY
```

Comment:
wowzers this looks like it's interacting with the default keosd? It needs to use its own isolated keosd.

Reply:
As above.  We're trying to be least intrusive with the backports unless we absolutely have to change something along the way, to try to maintain traceability.

Reply:
The goal of the backport effort isn't to _blindly_ take whatever was in block one's repo. This test clearly needs some refactoring to pass our quality standards. If you're concerned with traceability one option is to make another PR with changes that build on top of this PR.